python/ray/tune/trial.py

```diff
@@ -443,6 +445,22 @@ def last_result(self) -> dict:
     def last_result(self, val: dict):
         self._last_result = val
 
+    @property
+    def logdir(self):
+        if not self.relative_logdir:
+            return None
+        return str(Path(self.local_dir).joinpath(self.relative_logdir))
+
+    @logdir.setter
+    def logdir(self, logdir):
+        relative_logdir = Path(logdir).relative_to(self.local_dir)
+        assert str(relative_logdir).find("..") == -1
+        logger.warning(
+            "Deprecated. In future versions only the relative logdir "
+            "will be used and calling logdir will raise an error."
+        )
```

Comment:
Actually I think we can remove the deprecation warning for now. I think getting the absolute logdir as an utility is quite useful

Reply:
We should have this warning in the setter instead IMO

Reply:
It does make sense to keep it in the setter though, right?

Reply:
Yes, that makes sense! Please guard with `log_once` please